src/main/java/com/google/vinet/servlets/RegistrationServlet.java

```diff
@@ -0,0 +1,230 @@
+            System.err.printf("The %s is not formatted as a double!", propertyName);
+            return;
+          }
+        }
+
+        if (value == null) value = parameter;
+        entity.setProperty(propertyName, value);
+      }
+
+      datastore.put(entity);
+
+      response.setStatus(HttpServletResponse.SC_OK);
+
+      if (userType == UserType.ISOLATE) {
+        response.sendRedirect("/isolate/home.html");
+      } else if (userType == UserType.VOLUNTEER) {
+        response.sendRedirect("/volunteer/home.html");
+      } else {
+        response.setStatus(HttpServletResponse.SC_SERVICE_UNAVAILABLE);
+      }
+    } catch (Exception e) {
+      System.err.println("There was an error registering!");
+      response.setStatus(HttpServletResponse.SC_SERVICE_UNAVAILABLE);
+    }
+  }
+
+  /**
+   * Returns whether the logged-in User has registered with the service.
+   */
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException{
+    if (response == null) {
+      throw new IllegalArgumentException("response cannot be null");
+    }
+
+    if (request == null) {
+      throw new IllegalArgumentException("request cannot be null");
+    }
+
+    if ( !this.userService.isUserLoggedIn() ) {
+      response.sendError(HttpServletResponse.SC_UNAUTHORIZED);
+      return;
+    }
+
+    final User user = userService.getCurrentUser();
+    final String userId = user.getUserId();
+
+    if (userId == null) {
+      response.setStatus(HttpServletResponse.SC_UNAUTHORIZED);
+      return;
+    }
+
+    /* Create a query which will return all entries in the DataStore with a userId matching that of
+     *the current user. */
+    final Filter userIdFilter = new FilterPredicate("userId", FilterOperator.EQUAL, userId);
+    final Query query = new Query(USER_TABLE_NAME).setFilter(userIdFilter);
+
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    final PreparedQuery preparedQuery = datastore.prepare(query);
+
+    final Entity entity;
+
+    /* Try to retrieve the results of the query as a single Entity.
+     * If there is more than a single result, then the registration service has incorrectly
+     * registered a user twice. This error will be passed up to the caller.
+     *
+     * In a production environment, this would have to be reported to the sysadmin/project owner,
+     * as duplicate registration should never be allowed.
+     */
+    try{
+      entity = preparedQuery.asSingleEntity();
+    } catch(TooManyResultsException exception) {
+      response.sendError(HttpServletResponse.SC_INTERNAL_SERVER_ERROR);
+      throw exception;
+    } catch (IllegalStateException exception) {
+      response.sendError(HttpServletResponse.SC_INTERNAL_SERVER_ERROR);
+      throw exception;
+    }
+
+    final boolean registered;
+
+    if (entity == null) {
+      registered = false;
+    } else {
+      registered = true;
+    }
```

Comment:
`final boolean registered = (entity != null);`

Reply:
Oh that is a much nicer idiom for this, thank you!

Reply:
I've implemented this now https://github.com/googleinterns/step226-2020/pull/6/commits/8ab85b3931daea6bc79528f6fbcc086ed120a024